### ISSUE TITLE
feat: Ability to specify environment variables in config

### ISSUE THREAD

**Issue by theS1LV3R** [release-argus/Argus#362]
**Is your feature request related to a problem? Please describe.**
At the moment, Argus supports setting certain configuration elements through environment variables, e.g. `ARGUS_SERVICE_LATEST_VERSION_ACCESS_TOKEN` will set the `service.*.latest_version.access_token` config elements.

However this doesn't work very well for each service, for example I can't do `ARGUS_SERVICE_LOUISLAM_UPTIME_KUMA_DEPLOYED_VERSION_BASIC_AUTH_PASSWORD` to specify the value for `service.louislam/uptime-kuma.deployed_version.basic_auth.password`. (May be a bug, but I couldn't get it to work at least.)

**Describe the solution you'd like**
Instead of a mapping from environment variables to config properties, allow config properties to fetch from the environment. Then it would be possible to do something like this:

```yaml
service:
  louislam/uptime-kuma:
    deployed_version:
      basic_auth:
        password: $UPTIME_KUMA_PASSWORD
```

This could also be used with fallbacks the way bash does it, something like this:
```yaml
service:
  louislam/uptime-kuma:
    deployed_version:
      basic_auth:
        password: ${UPTIME_KUMA_PASSWORD:-some_default_password}
```

**Describe alternatives you've considered**
Not sure if there are any, seeing as hardcoding passwords in a git-committed file isn't a good idea exactly.

**Additional context**
None.

**Comment by JosephKav** [comment]
Alright, so I've looked into this a bit and what you're asking is possible. Fyi, `ARGUS_SERVICE_LATEST_VERSION_ACCESS_TOKEN` is the same as setting `defaults.service.latest_version.access_token` (though the file value would take precedence). There is currently only support for setting service value defaults via environment vars, not values for an individual service.

With what you're asking, I can't do it for non-strings without creating another var in all of these cases.
Strings on the other hand could be implemented far easier. The most lightweight would be to just enable it for things such as `latest_version.access_token`, `latest_version.require.docker.token`, `deployed_version.basic_auth.username`, `deployed_version.baaic_auth.password` and all `deployed_version.baaic_auth.headers.*.value`. Or, if you can give me a case for needing it on other string vars, I could look into making all string vars check whether they are an environment variable.

(Currently planning on making it `${ENV_VAR_NAME}` and regexing with `^\$\{(\w+)\}$`, then env lookup on that submatch)

**Comment by samcro1967** [comment]
I am missing something here as I have not been able to get this to work and am not seeing anything about it in the docs.  SO I have a file, .env, mounted to the container at /app/config/.env and then updated config.yml to use icon: "@TEST_ICON" in the dashboard section for an app.  Is this the correct way to use a .env file in Argus?

**Comment by JosephKav** [comment]
We don't currently have support for `.env` files inside Argus. You would have to get Docker Compose to pass these environment variables to the container (by setting them for the service e.g. `environment: - DEBUG=${DEBUG}`)
I could add support for reading in a `.env` to Argus if you'd like? (look for a `.env` file in the same dir as the config, and use it if present)
(and it'd be `${TEST_ICON}` to reference the `TEST_ICON` environment variable)

**Comment by samcro1967** [comment]
Apparently I misread this issue.  Thought it was for this specific ask.  Guess that is why I could not find it in the docs.  lol

Yes, would love support for an .env file.  Trying to avoid putting it in Docker compose as it will add a lot of lines.  Also trying to centralize settings acrcos multiple apps that use the same icons, URL links, etc.

**Comment by samcro1967** [comment]
@JosephKav I have this somewhat working now.  Below are the changes I made to the files so that argus can use a sourced .env file and then environment variables for icon: and icon_link_to: in the dashboard section for a service.   It also requires the .env file be attached to the docker container using `env_file: /path/to/.env`.  I could not figure out how to globally search and replace any variables in config.yml once it was loaded into memory so I focused on just the fields I was (selfishly) interested in.

I am not a developer so not proficient with go, git, github, and PRs, but hopefully this helps jump start adding this functionality to Argus.

[main.go.txt](https://github.com/user-attachments/files/19716365/main.go.txt)
[entrypoint.sh.txt](https://github.com/user-attachments/files/19716371/entrypoint.sh.txt)

**Comment by JosephKav** [comment]
Hey @samcro1967. I worked on this a bit tonight. Got it reading in environment variables from `.env` (if there's one in the dir with the config). It'd then use those when it resolves vars that support env var lookups.

I don't want to replace references to env vars in the config with the current values of the env var as that kinda forces you to keep a separate template copy of your config.

I'll add env var lookup support for those dashboard vars to my current PR for this (https://github.com/release-argus/Argus/pull/560) later in the week.

**Comment by JosephKav** [comment]
@samcro1967, this `.env` support should now be available in [0.25.0](https://github.com/release-argus/Argus/releases/tag/0.25.0). And `icon`, `icon_link_to` and `web_url` should all now support environment variables.
